exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/WorkspaceSchemaFactory.java

```diff
@@ -321,8 +327,101 @@ public DrillTable create(String key) {
       return null;
     }
 
+    private FormatMatcher findMatcher(FileStatus file) {
+      FormatMatcher matcher = null;
+      try {
+        for (FormatMatcher m : dropFileMatchers) {
+          if (m.isFileReadable(fs, file)) {
+            return m;
+          }
+        }
+      } catch (IOException e) {
+        logger.debug("Failed to find format matcher for file: %s", file, e);
+      }
+      return matcher;
+    }
+
     @Override
     public void destroy(DrillTable value) {
     }
+
+    /**
+     * Check if the table contains homogenenous files that can be read by Drill. Eg: parquet, json csv etc.
+     * However if it contains more than one of these formats or a totally different file format that Drill cannot
+     * understand then we will raise an exception.
+     * @param key
+     * @return
+     * @throws IOException
+     */
+    private boolean isHomogeneous(String key) throws IOException {
```

Comment:
The only reason was to avoid the performance penalty for read that would be ensued by these checks and simply going ahead optimistically and failing later if we hit different formats.